src/components/manage/Form/Field.jsx

```diff
@@ -155,9 +177,9 @@ const UnconnectedField = (props, { intl }) => {
     getWidgetByFieldId(props.id) ||
     getWidgetFromTaggedValues(props.widgetOptions) ||
     getWidgetByName(props.widget) ||
-    getWidgetByChoices(props) ||
     getWidgetByVocabulary(props.vocabulary) ||
     getWidgetByVocabularyFromHint(props) ||
+    getWidgetByChoices(props) ||
```

Comment:
This, unfortunately, it's a breaking change. What is the reason behind?

Reply:
Check the getWidgetByChoices implementation; it checks if there is a vocabulary and defaults to a choices widget, even if the vocabulary is the catalog one. So that is why the vocabulary should be checked first.

Reply:
Let's do it then, seems safe.